clients/drcachesim/tracer/raw2trace.cpp

```diff
@@ -47,7 +47,17 @@
 #include "utils.h"
 #ifdef LINUX
 // XXX: We should have the core export this to an include dir.
-#    include "../../core/unix/include/syscall.h"
+#    ifdef X86
+#        include "../../core/unix/include/syscall_linux_x86.h"
```

Comment:
Xref @derekbruening's comment on #6406: https://github.com/DynamoRIO/dynamorio/issues/6406#issuecomment-1793322402

(Moving the discussion here)

If host!=target, for raw2trace it seems correct to me that we would always want to use the target arch's system calls (like we use the target arch's decoder). In what case would we need the host's?

Reply:
For raw2trace, yes, we always want the target.  But for core DR even when just used for decoding it might do printing to the console or other actions using direct syscalls (if the same executable is using full DR statically linked and so can't easily switch to drdecodelib) and there it wants host syscalls.  So the picture is complex for core DR.